### PR TITLE
Parse size use, etc

### PR DESCRIPTION
**Pull request by NiceneNerd**
- Removes `_FactoryParser` class
- Use parse size values
- Add 8 bytes to Wii U parse size for `.hkrb`

### CODE REVIEW

**Comment by leoetlino** [comment]
Are you sure Nintendo adds 32 bytes for all files? That seems wrong for simple resources like BYML.

**Comment by NiceneNerd** [comment]
You know, I meant to run some tests to make sure of that on other types. I'll get back to you in a little bit.

File: rstb/rstb.py [hunk 1]
```diff
@@ -193,12 +184,7 @@ def calculate_file_size_with_ext(self, file: typing.Union[str, os.PathLike, typi
         if wiiu:
             size += 0xe4 # res::ResourceMgr constant. Not sure what it is.
             size += info.size_wiiu
-            if actual_ext not in _factory_parsers:
-                size += info.parse_size_wiiu
-            elif info.is_complex:
-                size += _factory_parsers[actual_ext].parse_wiiu(file_data)
-            else:
-                size += 0
+            size += info.parse_size_wiiu
```
Comment (leoetlino) [inline rstb/rstb.py:186]: The is_complex check needs to be kept, otherwise you'll be adding "complex" to `size` for complex resource classes.

Comment (NiceneNerd) [reply rstb/rstb.py:186]: Don't the parse size values on `Factory` not even get set when a type is complex?
https://github.com/zeldamods/rstb/blob/1fdea29eecd7096a999de40f81d9b587959985a7/rstb/rstb.py#L144

I would assume this leaves them as 0.

Comment (leoetlino) [reply rstb/rstb.py:186]: Oh, my bad, I forgot about that check.

**Comment by NiceneNerd** [comment]
> Are you sure Nintendo adds 32 bytes for all files? That seems wrong for simple resources like BYML.

Okay, so this definitely does not occur with BYML. That's quite curious. Is there a chance it rounds differently for different files? Because the +32 for `.hkrb` file sizes that are divisible by 32 does match the data. Should there just be special handling? Or could it also apply to other Havok stuff?

**Comment by NiceneNerd** [comment]
On another note, I wonder if not using parse size values before is what caused problems with `.bdmgparam`, being the only non-complex type in the table with a parse size greater than 0.

**Comment by leoetlino** [comment]
Yeah, it's possible rkrb are handled differently. I'd just always add 32 as parse_size to keep it simple.

**Comment by NiceneNerd** [comment]
32 or 40? Because I already added 8 into the parse size for those that are already too small but aren't divisible by 32, and those which are were originally 40 short.

**Comment by leoetlino** [comment]
I meant add another 32 bytes, which would make the final parse size 40 I guess.

**Comment by NiceneNerd** [comment]
Oh, okay, that makes sense.

**Pull request merged**